cpp/src/command_classes/CentralScene.cpp

```diff
@@ -162,34 +181,48 @@ bool CentralScene::HandleMsg
 	if( CentralSceneCmd_Set == (CentralSceneCmd)_data[0] )
 	{
 		// Central Scene Set received so send notification
-		int32 when;
-		if( _data[2] == 0 )
-			when = 0;
-		else if( _data[2] <= 0x7F )
-			when = _data[2];
-		else if( _data[2] <= 0xFE )
-			when = 60 * _data[2];
-		else
-			when = 0;
-		Log::Write( LogLevel_Info, GetNodeId(), "Received Central Scene set from node %d: scene id=%d in %d seconds. Sending event notification.", GetNodeId(), _data[3], when);
-
-		if( ValueInt* value = static_cast<ValueInt*>( GetValue( _instance, _data[3] ) ) )
+		uint8 keyAttribute = _data[2];
+		uint8 sceneID = _data[3];
+		Log::Write( LogLevel_Info, GetNodeId(), "Received Central Scene set from node %d: scene id=%d with key Attribute %d. Sending event notification.", GetNodeId(), sceneID, keyAttribute);
+
+		int index = 1;
+		if      ( keyAttribute == 0) { index = CentralScene_SinglePress; }
+		else if ( keyAttribute == 1) { index = CentralScene_Released; }
+		else if ( keyAttribute == 2) { index = CentralScene_HeldDown; }
+		else if ( keyAttribute == 3) { index = CentralScene_DoublePress; }
+		else if ( keyAttribute == 4) { index = CentralScene_TriplePress; }
+		else if ( keyAttribute == 5) { index = CentralScene_Pressed4Times; }
+		else if ( keyAttribute == 6) { index = CentralScene_Pressed5Times;}
+
+		if( ValueInt* value = static_cast<ValueInt*>( GetValue( sceneID, index ) ) )
 		{
-			value->OnValueRefreshed( when );
+			value->OnValueRefreshed( 1 );
 			value->Release();
 		} else {
 			Log::Write( LogLevel_Warning, GetNodeId(), "No ValueID created for Scene %d", _data[3]);
 			return false;
 		}
 		return true;
-	} else if (CentralSceneCmd_Capability_Report == (CentralSceneCmd)_data[0]) {
+	}
+	else if (CentralSceneCmd_Capability_Report == (CentralSceneCmd)_data[0])
+	{
 		/* Create a Number of ValueID's based on the m_scenecount variable
 		 * We prefer what the Config File specifies rather than what is returned by
 		 * the Device...
 		 */
 		int scenecount = _data[1];
-		if (m_scenecount != 0)
+		if (scenecount != 0)
+		{
 			m_scenecount = scenecount;
+		}
+		bool identical = false;
+		bool version2orHigher = false;
+		if ( _length >= 2 )
```

Comment:
yes you are right, but for some reason i don't get the version for CENTRAL_SCENE. So this was just a dirty workaround.
```
2016-09-28 11:22:49.774 Info, Node017, Requesting Versions for COMMAND_CLASS_CRC_16_ENCAP
2016-09-28 11:22:49.774 Info, Node017, Requesting Versions for COMMAND_CLASS_DEVICE_RESET_LOCALLY
2016-09-28 11:22:49.774 Info, Node017, Requesting Versions for COMMAND_CLASS_CENTRAL_SCENE
2016-09-28 11:22:49.774 Info, Node017, Requesting Versions for COMMAND_CLASS_ZWAVE_PLUS_INFO
2016-09-28 11:22:49.774 Info, Node017, Requesting Versions for COMMAND_CLASS_CONFIGURATION
2016-09-28 11:22:49.774 Info, Node017, Requesting Versions for COMMAND_CLASS_ALARM
2016-09-28 11:22:49.774 Info, Node017, 	ok
2016-09-28 11:22:49.774 Detail, Node017, Setting Encryption Flag on Message For Command Class COMMAND_CLASS_VERSION
2016-09-28 11:22:49.774 Detail, Node017, Queuing (Send) VersionCmd_CommandClassGet (Node=17): 0x01, 0x0a, 0x00, 0x13, 0x11, 0x03, 0x86, 0x13, 0x71, 0x25, 0x30, 0x05
2016-09-28 11:22:49.774 Info, Node017, Requesting Versions for COMMAND_CLASS_MANUFACTURER_SPECIFIC
...
```
I just get the versions for SWITCH_MULTILEVE, ALARM and WAKEUP.
Do i need to set something, so that openZwave asks for the Version of CENTRAL_SCENE?